web_external/views/body/SubmitView.js

```diff
@@ -34,6 +35,10 @@ var SubmitView = View.extend({
         'input .c-submission-documentation-url-input': function (event) {
             this.documentationUrl = $(event.currentTarget).val().trim();
             this.validateInputs();
+        },
+        'input .c-submission-approach-input': function (event) {
+            this.submission.set('approach', $(event.currentTarget).val().trim() || 'default');
```

Comment:
For me, this method isn't called when I select a completion from the typeahead popup menu. Therefore, even though the value in the text field is updated with the selected completion, the `approach` attribute isn't set on the Submission model, and the wrong value is POSTed.

Reply:
Using the submission model for this was overly complicated, so I changed it to store the approach on the view.  I think it works correctly, but let me know if you see any bugs.